dashboard/src/components/TestsTable/TestsTable.tsx

```diff
@@ -107,39 +145,69 @@ export function TestsTable({
           hardware: buildHardwareArray(e.environment_compatible, e.misc),
           treeBranch: buildTreeBranch(e.tree_name, e.git_repository_branch),
         });
-        switch (e.status) {
-          case 'DONE':
-            groups[group].done_tests++;
-            break;
-          case 'ERROR':
-            groups[group].error_tests++;
-            break;
-          case 'FAIL':
-            groups[group].fail_tests++;
-            break;
-          case 'MISS':
-            groups[group].miss_tests++;
-            break;
-          case 'PASS':
-            groups[group].pass_tests++;
-            break;
-          case 'SKIP':
-            groups[group].skip_tests++;
-            break;
-          default:
-            groups[group].null_tests++;
-        }
       });
     }
     return Object.values(groups);
   }, [testHistory]);
 
+  const [globalStatusGroup, pathFilteredData] = useMemo((): [
+    TPathTestsStatus,
+    TPathTests[],
+  ] => {
+    const path = globalFilter;
+    const isValidPath = path !== undefined && path !== '';
+    const globalGroup: TPathTestsStatus = {
+      done_tests: 0,
+      fail_tests: 0,
+      miss_tests: 0,
+      pass_tests: 0,
+      null_tests: 0,
+      skip_tests: 0,
+      error_tests: 0,
+      total_tests: 0,
+    };
+
+    const filteredData = rawData.reduce<TPathTests[]>((acc, test) => {
+      const localGroup: TPathTestsStatus = {
+        done_tests: 0,
+        fail_tests: 0,
+        miss_tests: 0,
+        pass_tests: 0,
+        null_tests: 0,
+        skip_tests: 0,
+        error_tests: 0,
+        total_tests: 0,
+      };
+      const individualTest = test.individual_tests.filter(t => {
+        let dataIncludesPath = true;
+        if (isValidPath) dataIncludesPath = t.path.includes(path);
```

Comment:
nit:
```suggestion
        if (isValidPath) {
         dataIncludesPath = t.path.includes(path);
         }
```
we should probably add this to eslint